src/rgw/radosgw-admin/radosgw-admin.cc

```diff
@@ -7952,6 +7957,50 @@ int main(int argc, const char **argv)
     return 0;
   }
 
+  if (opt_cmd == OPT::BUCKET_LOGGING_LIST) {
+    if (bucket_name.empty()) {
+      cerr << "ERROR: bucket not specified" << std::endl;
+      return EINVAL;
+    }
+    int ret = init_bucket(tenant, bucket_name, bucket_id, &bucket);
+    if (ret < 0) {
+      return -ret;
+    }
+
+    rgw::bucketlogging::configuration configuration;
+    std::unique_ptr<rgw::sal::Bucket> target_bucket;
+    ret =  rgw::bucketlogging::get_target_and_conf_from_source(dpp(),
+         driver, bucket.get(), tenant, configuration, target_bucket, null_yield);
+    if (ret < 0 && ret != -ENODATA) {
+      cerr << "ERROR: failed to get target bucket and logging conf from source bucket '"
+        << bucket_name << "': " << cpp_strerror(-ret) << std::endl;
+      return -ret;
+    } else if (ret == -ENODATA) {
+      cerr << "ERROR: bucket '" << bucket_name << "' does not have logging enabled" << std::endl;
+      return 0;
+    }
+    std::string target_prefix = configuration.target_prefix;
+    std::set<std::string> entries;
+
+    ret = rgw::bucketlogging::list_pending_commit_objects(dpp(),
+        static_cast<rgw::sal::RadosStore*>(driver), target_bucket.get(),
```

Comment:
why is this needed?
can you just pass `driver`?

Reply:
Can you please elaborate?

Reply:
my mistake. i thought that this is a generic API, not rados sepcific.